rudder-core/src/main/scala/com/normation/rudder/domain/reports/ExpectedReports.scala

```diff
@@ -44,8 +44,25 @@ import com.normation.inventory.domain.NodeId
 
 
 /**
+ * The main class helping maps expected
+ * reports for a node
+ */
+case class RuleNodeExpectedReports(
```

Comment:
It's a datastructure above node. In the same idea that components don't have reference to directive, that datastracture does not have reference to node. 
The actual full structure at top level looks like: Map[NodeId, Map[NodeConfigId, Set[RuleNodeExpectedReports]]

So, we could add VersionnedNodeExpectedReports with a config Id, but I didn't feel the need in the code. Perhaps it's a fault.